### PR TITLE
Horizon sensor console announcement of incoming fire

### PR DESCRIPTION
**Pull request by FluffyGhoster**
Added ability to the sensors console to send an announcement about incoming fire, along with the direction it's coming from, and to announce when there's no fire incoming (and thus to resume damage control.

The panel only appears on the horizon itself (or a ship that spoofs its id_name, possible play for antags and also because i didn't see any better way to do it)

### CODE REVIEW

**Comment by FluffyGhoster** [comment]
!review

**Review by JohnWildkins — APPROVED** [review]

File: code/modules/overmap/ships/computers/sensors.dm [hunk 1]
```diff
@@ -183,6 +183,15 @@
 		visible_message(SPAN_ITALIC("[accent_icon] <b>[user_name]</b> explains, \"[beacon.distress_message]\""))
 		return TOPIC_HANDLED
 
+	if(href_list["inbound_fire"])
+		var/direction = href_list["inbound_fire"]
+		if(direction != "clear")
+			security_announcement.Announce("Enemy fire inbound, enemy fire inbound! [direction]!", "Brace for shock!", sound('sound/mecha/internaldmgalarm.ogg'), 0)
+		else
+			security_announcement.Announce("No fire is incoming at the current moment, resume damage control", "Space clear!", sound('sound/misc/announcements/security_level_old.ogg'), 0)
```
Comment (NonQueueingMatt) [inline code/modules/overmap/ships/computers/sensors.dm:191]:
```suggestion
			security_announcement.Announce("No fire is incoming at the current moment, resume damage control.", "Space clear!", sound('sound/misc/announcements/security_level_old.ogg'), 0)
```

**Review by NonQueueingMatt — APPROVED** [review]

**Comment by SleepyGemmy** [comment]
DNM while i playtest.

File: nano/templates/shipsensors.tmpl [hunk 2]
```diff
@@ -119,6 +119,31 @@
...
+			{{:helper.link('', 'triangle-1-e', { 'inbound_fire' : 'Starboard' }, null, null)}}
+		</div>
+		<div class='item'>
+			{{:helper.link('', 'triangle-1-sw', { 'inbound_fire' : 'Bow-Port' }, null, null)}}
+			{{:helper.link('', 'triangle-1-s', { 'inbound_fire' : 'Bow' }, null, null)}}
+			{{:helper.link('', 'triangle-1-se', { 'inbound_fire' : 'Bow-Starboard' }, null, null)}}
+		</div>
+		</fieldset>
+</div>
+{{/if}}
```
Comment (SleepyGemmy) [inline nano/templates/shipsensors.tmpl:145]:
```suggestion
{{if data.id_name == 'Horizon'}}
<h4>Announce Incoming Projectiles</h4>
<div class='block'>
	<div class='item'>
		<div class='item'>
			{{:helper.link('', 'triangle-1-nw', { 'inbound_fire' : 'fore port' }, null, null)}}
			{{:helper.link('', 'triangle-1-n', { 'inbound_fire' : 'fore' }, null, null)}}
			{{:helper.link('', 'triangle-1-ne', { 'inbound_fire' : 'fore starboard' }, null, null)}}
		</div>
		<div class='item'>
			{{:helper.link('', 'triangle-1-w', { 'inbound_fire' : 'port' }, null, null)}}
			{{:helper.link('', 'circle-close', { 'inbound_fire' : 'clear' }, null, null)}}
			{{:helper.link('', 'triangle-1-e', { 'inbound_fire' : 'starboard' }, null, null)}}
		</div>
		<div class='item'>
			{{:helper.link('', 'triangle-1-sw', { 'inbound_fire' : 'aft port' }, null, null)}}
			{{:helper.link('', 'triangle-1-s', { 'inbound_fire' : 'aft' }, null, null)}}
			{{:helper.link('', 'triangle-1-se', { 'inbound_fire' : 'aft starboard' }, null, null)}}
		</div>
	</div>
</div>
{{/if}}
```
looks like this:
image

**Review by SleepyGemmy — DISMISSED** [review]

File: code/modules/overmap/ships/computers/sensors.dm [hunk 3]
```diff
@@ -183,6 +183,15 @@
 		visible_message(SPAN_ITALIC("[accent_icon] <b>[user_name]</b> explains, \"[beacon.distress_message]\""))
 		return TOPIC_HANDLED
 
+	if(href_list["inbound_fire"])
+		var/direction = href_list["inbound_fire"]
+		if(direction != "clear")
+			security_announcement.Announce("Enemy fire inbound, enemy fire inbound! [direction]!", "Brace for shock!", sound('sound/mecha/internaldmgalarm.ogg'), 0)
```
[the remark this answers is not in the document]
Comment (FluffyGhoster) [reply code/modules/overmap/ships/computers/sensors.dm:189]: The volume can be tweaked in the last param of `sound` (though i didn't find it too loud), I do not like the idea of announcing a "projectile" as it's not limited to those when more things will be added (eg, torpedoes, beams, whatnot)

[the remark this answers is not in the document]
Comment (SleepyGemmy) [reply code/modules/overmap/ships/computers/sensors.dm:189]: anything fired from a weapon is a projectile.

[the remark this answers is not in the document]
Comment (FluffyGhoster) [reply code/modules/overmap/ships/computers/sensors.dm:189]: On the code, yes, but that's not what most people think when you say projectile, it's also not how it's usually announced (see https://youtu.be/yonyV6_6ur4?t=160 for example), it's announced as "[type] inbound, [type] inbound! Brace for shock!" (yes a missle is also a projectile, but it's not called as such)

File: nano/templates/shipsensors.tmpl (hunk 2 above)
Comment (FluffyGhoster) [reply nano/templates/shipsensors.tmpl]: implemented new style suggestion in commit [0e6bc19](https://github.com/Aurorastation/Aurora.3/pull/15117/commits/0e6bc19af614dd7a1c8b2d3103c744fd24f1279a)

File: code/modules/overmap/ships/computers/sensors.dm (hunk 3 above)
[the remark this answers is not in the document]
Comment (FluffyGhoster) [reply code/modules/overmap/ships/computers/sensors.dm]: Implemented sound volume reduction for the internaldmgalarm in [0e6bc19](https://github.com/Aurorastation/Aurora.3/pull/15117/commits/0e6bc19af614dd7a1c8b2d3103c744fd24f1279a)

**Pull request merged**